frappe/model/document.py

```diff
@@ -488,6 +489,15 @@ def set_docstatus(self):
 		for d in self.get_all_children():
 			d.docstatus = self.docstatus
 
+	def set_owner(self):
+		"""Override the owner if owner is not a session user"""
+		if self.is_new():
+			self.owner = self.flags.owner or self.modified_by
+		else:
+			doc_before_save = self.get_doc_before_save()
+			if doc_before_save.owner != self.owner:
+				self.owner = self.flags.owner or doc_before_save.owner
```

Comment:
This should raise instead of handling it silently IMO

Reply:
Raising an exception might break many other test cases since it is very tightly coupled with todos created via assignment.